cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -401,6 +403,55 @@ public synchronized void updateProperties(NamespaceId namespaceId, NamespaceMeta
       throw new BadRequestException(String.format("Mappings %s for namespace %s cannot be updated once the namespace " +
                                                     "is created.", difference, namespaceId));
     }
+
+    if (namespaceMeta.getRepoConfig() != null) {
+      updateRepoConfig(builder, namespaceMeta.getRepoConfig());
+    }
+    
+    NamespaceMeta updatedMeta = builder.build();
+    nsStore.update(updatedMeta);
+    // refresh the cache with new meta
+    namespaceMetaCache.refresh(namespaceId);
+    LOG.info("Namespace {} updated with meta {}", namespaceId, updatedMeta);
+  }
+
+  private void updateRepoConfig(NamespaceMeta.Builder builder, NamespaceRepositoryConfig repoConfig) {
+    if (repoConfig.getProvider() != null) {
+      builder.setRepoProvider(repoConfig.getProvider());
+    }
+    if (repoConfig.getLink() != null) {
+      builder.setRepoLink(repoConfig.getLink());
+    }
+    if (repoConfig.getDefaultBranch() != null) {
+      builder.setRepoDefaultBranch(repoConfig.getDefaultBranch());
+    }
+    if (repoConfig.getAuthType() != null) {
+      builder.setRepoAuthType(repoConfig.getAuthType());
+    }
+    if (repoConfig.getUsername() != null) {
+      builder.setRepoUserName(repoConfig.getUsername());
+    }
+    if (repoConfig.getPathPrefix() != null) {
+      builder.setRepoPathPrefix(repoConfig.getPathPrefix());
+    }
+  }
+
+  @Override
+  public synchronized void deleteRepository(NamespaceId namespaceId) throws Exception {
```

Comment:
This class has many public methods marked with `synchronized`, which I believe is unnecessary. Can you check with @albertshau on this?

Reply:
hmmm, I followed existing PUT method pattern, putting it as synchronized because it's fetch and update `NamespaceMeta`, as you commented below, does it make sense?

Reply:
Does this ticket still apply: https://cdap.atlassian.net/browse/CDAP-1427? It's one of the TODOs in this class

Reply:
+1, these should not be relying on synchronized, but on performing the needed operations within a single transaction

Reply:
For the added new methods, I've put them in one transaction. For the existing synchronized methods that deal with external calls like the delete namespace, which cannot be handled in one transaction yet, I left them untouched. Is it okay?